pkg/local_object_storage/metabase/metadata.go

```diff
@@ -111,6 +110,32 @@ func putMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID, ver version.Version, owner
 	return nil
 }
 
+func deleteMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID) error {
+	metaBkt := tx.Bucket(metaBucketKey(cnr))
+	if metaBkt == nil {
+		return nil
+	}
+	// removed keys must be pre-collected according to BoltDB docs.
+	// TODO: consider calculating key set from the object header
+	ks := make([][]byte, 0, 10) // at least system attributes (approx)
+	c := metaBkt.Cursor()
+	for k, _ := c.First(); k != nil; k, _ = c.Next() {
```

Comment:
It's horribly expensive and redundant to me. You have `metaPrefixIDAttr` data that can be used to find all the other keys.

Reply:
sure, done